src/Orleans.Core/Statistics/ClientStatisticsManager.cs

```diff
@@ -56,18 +59,18 @@ internal class ClientStatisticsManager : IDisposable
                 if (configurableMetricsDataPublisher != null)
                 {
                     configurableMetricsDataPublisher.AddConfiguration(
-                        this.clusterClientOptions.ClusterId, config.DNSHostName, clientId.ToString(), transport.MyAddress.Endpoint.Address);
+                        this.clusterClientOptions.ClusterId, dnsHostName, clientId.ToString(), transport.MyAddress.Endpoint.Address);
                 }
                 tableStatistics = new ClientTableStatistics(transport, metricsDataPublisher, this.hostEnvironmentStatistics, this.appEnvironmentStatistics, this.loggerFactory)
                 {
                     MetricsTableWriteInterval = statisticsOptions.MetricsTableWriteInterval
                 };
             }
-            else if (config.UseAzureSystemStore)
+            else if (CanUseAzureTable())
```

Comment:
Can we avoid this altogether?

Reply:
I'd love to. But I think refactor statistic system is out of the scope of this PR. We can refactor it in another pr , with refactoring silo statistics too.

Reply:
This feels hacky, but as long as there's an issue opened for refactoring it before 2.0

Reply:
yeah. I will open one if there isn't one.

Reply:
it was hacky before ;) this pr didn't made it worse. yeah But we should refactor it.

Reply:
opened an issue in #3976